File: 1128. Number of Equivalent Domino Pairs/solution.js
/**
 * @param {number[][]} dominoes
 * @return {number}
 */
 var numEquivDominoPairs = function(dominoes) {
  let myDict = {};
  let result = 0;
  for (let domino of dominoes){
    let usedDomino = domino[0].toString() + domino[1].toString();
    let flippedDomino = domino[1].toString() + domino[0].toString();
    if (myDict[flippedDomino]){
      usedDomino = flippedDomino;
    }
    if (myDict[usedDomino]){
      result += myDict[usedDomino];
      myDict[usedDomino] += 1;
    }
    else{
      myDict[usedDomino] = 1;
    }
  }

  return result;
};

File: 1128. Number of Equivalent Domino Pairs/solution.py
class Solution(object):
  def numEquivDominoPairs(self, dominoes):
    """
    :type dominoes: List[List[int]]
    :rtype: int
    """
    myDict = {}
    result = 0
    for domino in dominoes:
      usedDomino = str(domino[0]) + str(domino[1])
      flippedDomino = str(domino[1]) + str(domino[0])
      if flippedDomino in myDict:
        usedDomino = flippedDomino
      if usedDomino in myDict:
        result += myDict[usedDomino]
        myDict[usedDomino] += 1
      else:
        myDict[usedDomino] = 1
    
    return result